libasynql/src/poggit/libasynql/DataConnector.php

```diff
@@ -120,6 +120,13 @@ public function executeSelectRaw(string $query, array $args = [], ?callable $onS
 	 */
 	public function waitAll() : void;
 
+	/**
+	 * This function checks for the results of the queries handled before it was called. and proceeds to run them... check any implementation of SqlThread->readResults()
+	 *
+	 * @internal
```

Comment:
If it is an internal method, it shouldn't be in this interface.
Internals can always take a DataConnectorImpl directly. The interface is only intended for dependency injection in the userland.